test/Main.purs

```diff
@@ -0,0 +1,123 @@
+import OIDCCryptUtils.JSONWebKey (JSONWebKey())
+import Data.Either.Unsafe (fromRight)
+import Data.Argonaut.Decode (decodeJson)
+import Data.Argonaut.Parser (jsonParser)
+
+-- Token generated using https://jwt.io/
+-- JWK generated from public PEM using https://www.npmjs.com/package/pem-jwk
+-- idToken (and therefore tests) valid until 2050 ; )
+idToken :: String
+idToken = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiI5NDU0NzkyNjQyMzMta3Vtc2tvMHEzZTVlaDNlZmQ4cGxsMWRhOWt1YTNpM2IuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJhenAiOiI5NDU0NzkyNjQyMzMta3Vtc2tvMHEzZTVlaDNlZmQ4cGxsMWRhOWt1YTNpM2IuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJlbWFpbCI6ImJlY2t5QHNsYW1kYXRhLmNvbSIsImlzcyI6Imh0dHBzOi8vYWNjb3VudHMuZ29vZ2xlLmNvbSIsIm5vbmNlIjoiNWQ0MTQwMmFiYzRiMmE3NmI5NzE5ZDkxMTAxN2M1OTIiLCJleHAiOjI1MjQ2MDgwMDAsImlhdCI6MTQ1NTA2OTM0OCwic3ViIjoiMTA3NjEwNzc2OTUxMDk4NjQzOTUwIn0.PUi5wySBaUSBt9lepycGton2_-plIaX14q19NB13GF8ISa9gUwnt4LeHWPst42jBeAO1GJ_thIYm6gQIIKBMtr0hucddfzu7oWXfobeFke-WwBHgmFIKSccWhI-QoNxmDbJkNolo_oPsu3DcOpFHKrnmrTWuQFZpdYciGpYi72k"
+
+idTokenWithMultipleAudiences :: String
+idTokenWithMultipleAudiences = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsiYXVkMSIsImF1ZDIiXSwiYXpwIjoiYXVkMSIsImVtYWlsIjoiYmVja3lAc2xhbWRhdGEuY29tIiwiaXNzIjoiaHR0cHM6Ly9hY2NvdW50cy5nb29nbGUuY29tIiwibm9uY2UiOiI1ZDQxNDAyYWJjNGIyYTc2Yjk3MTlkOTExMDE3YzU5MiIsImV4cCI6MjUyNDYwODAwMCwiaWF0IjoxNDU1MDY5MzQ4LCJzdWIiOiIxMDc2MTA3NzY5NTEwOTg2NDM5NTAifQ.O0bLeKAvB9e22nP_QKuyMnMKBrKDfwbzamgPxc_JXYLoF6cR-goXjPEZytfSR59YZAj6vsVZ-QwTFIY51tGk3qsCwvZw2ynRpMLcjf91FUmNoQvFcDn6Ltp_G9SWcQ_MsWnP9s8WD7AstCUa7E6VSe9nD2Zyazb9BD3iTYzhIrU"
+
+idTokenWithMultipleAudiencesWhichDontMatchTheAzp :: String
+idTokenWithMultipleAudiencesWhichDontMatchTheAzp = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsiYXVkMSIsImF1ZDIiXSwiYXpwIjoiYXVkMyIsImVtYWlsIjoiYmVja3lAc2xhbWRhdGEuY29tIiwiaXNzIjoiaHR0cHM6Ly9hY2NvdW50cy5nb29nbGUuY29tIiwibm9uY2UiOiI1ZDQxNDAyYWJjNGIyYTc2Yjk3MTlkOTExMDE3YzU5MiIsImV4cCI6MjUyNDYwODAwMCwiaWF0IjoxNDU1MDY5MzQ4LCJzdWIiOiIxMDc2MTA3NzY5NTEwOTg2NDM5NTAifQ.Cs10otwS9hvhYgBZ8yK8Au9Aki7r9OQIFv8o9p9zflPygulaaeLetA3EpJGey9Zm4vyB7hoqhhTx0-2Lv3cop22dp_Ne-X2BF-JoaiddI8I-kVwpm_-hBE2LW3EmGttFgSRZp70-iH65PP1HOiUz1abU8hpfk0VYLjpEJPadYyQ"
+
+azp :: String
+azp = "aud1"
+
+state :: String
+state = "https://github.com"
+
+stateKey :: String
+stateKey = "7cb76d4fa8a18c33906a7d03c6d7d627"
+
+weirdIdToken :: String
+weirdIdToken = "70eec61e97a9b0c33abfbd554ae5556a"
+
+issuer :: String
+issuer = "https://accounts.google.com"
+
+clientId :: String
+clientId = "945479264233-kumsko0q3e5eh3efd8pll1da9kua3i3b.apps.googleusercontent.com"
+
+nonce :: String
+nonce = "hello"
+
+publicJWKString :: String
+publicJWKString =
+  "{ \"kty\": \"RSA\", \"alg\": \"RS256\", \"use\": \"sig\", \"kid\": \"1\", \"n\": \"3ZWrUY0Y6IKN1qI4BhxR2C7oHVFgGPYkd38uGq1jQNSqEvJFcN93CYm16_G78FAFKWqwsJb3Wx-nbxDn6LtP4AhULB1H0K0g7_jLklDAHvI8yhOKlvoyvsUFPWtNxlJyh5JJXvkNKV_4Oo12e69f8QCuQ6NpEPl-cSvXIqUYBCs\", \"e\": \"AQAB\"}"
```

Comment:
Thats a good idea but I'd rather leave this for now.

The escaped JSON strings are what you get out of the tools used to generate these JWKs and its very hard  to construct them arbitrarily.